### PR TITLE
TRT-1545: Begin removing use of the legacy locator on intervals

### PR DESCRIPTION
**Pull request by dgoodwin**
This is continued work on the structured intervals, everything has been ported to the new format but before we can remove the legacy "locator" field we need the code to stop using it. This is just a small start on porting some of those to use StructuredInterval.

### CODE REVIEW

File: pkg/monitortests/node/nodestateanalyzer/nodeTest/drain-02/startingEvents.json [hunk 1]
```diff
@@ -2,31 +2,101 @@
   "items": [
     {
       "level": "Info",
-      "locator": "ns/openshift-machine-config-operator node/ci-op-wh7nq7n4-206af-q9jvw-worker-a-rdxtm",
+      "locator": "ns/openshift-machine-config-operator node/ci-op-wh7nq7n4-206af-q9jvw-worker-a-rdxtm hmsg/94b2a62d8a",
```
Comment (DennisPeriquet) [inline pkg/monitortests/node/nodestateanalyzer/nodeTest/drain-02/startingEvents.json:5]: I don't know if it matters, but, the 3 other additions to this file also have a locator but they don't contain the `hmsg/.....`.

File: pkg/monitortests/testframework/disruptionserializer/write_job_run_data_test.go [hunk 2]
```diff
@@ -45,30 +60,75 @@ func TestComputeDisruptionData(t *testing.T) {
...
 					Condition: monitorapi.Condition{
-						Level:   monitorapi.Error,
-						Locator: "backend-disruption-name/kube-api-new-connections disruption/kube-api connection/new",
-						Message: "reason/DisruptionBegan ns/openshift-image-registry route/test-disruption-new disruption/image-registry connection/new stopped responding to GET requests over new connections: Get \"https://test-disruption-new-openshift-image-registry.apps.ci-op-4mb5069g-03fd1.XXXXXXXXXXXXXXXXXXXXXX/healthz\": dial tcp 104.154.53.8:443: connect: connection refused",
+						Level: monitorapi.Error,
+						StructuredLocator: monitorapi.Locator{
+							Type: monitorapi.LocatorTypeDisruption,
+							Keys: map[monitorapi.LocatorKey]string{
+								monitorapi.LocatorBackendDisruptionNameKey: "kube-api-new-connections",
+								monitorapi.LocatorDisruptionKey:            "kube-api",
```
Comment (DennisPeriquet) [inline pkg/monitortests/testframework/disruptionserializer/write_job_run_data_test.go:92]: We set `LocatorDisruptionKey` to `kube-api` here, but the original `Message` says `disruption/image-registry` -- maybe the original was wrong and should've said `disruption/kube-api`?

lines 177, 225, 310, 382 (with `openshift-api`) also have this inconsistency.

**Review by DennisPeriquet — COMMENTED** [review]
Just a few minor questions on the changes to the tests.  Otherwise, looks ready.

**Comment by DennisPeriquet** [comment]
/lgtm
/hold

if my comments are ok, you can hold cancel anytime.

**Comment by dgoodwin** [comment]
/hold cancel

Large chunk of work ready to go that depends on this after shift week.

**Comment by dgoodwin** [comment]
```
[sig-arch] events should not repeat pathologically for ns/openshift-kube-apiserver-operator expand_less 	0s
{  1 events happened too frequently

event happened 22 times, something is wrong: namespace/openshift-kube-apiserver-operator deployment/kube-apiserver-operator hmsg/7c79ac0a72 - reason/RequiredInstallerResourcesMissing configmaps: aggregator-client-ca,check-endpoints-kubeconfig,client-ca,control-plane-node-kubeconfig, configmaps: bound-sa-token-signing-certs-9,config-9,etcd-serving-ca-9,kube-apiserver-audit-policies-9,kube-apiserver-cert-syncer-kubeconfig-9,kube-apiserver-pod-9,kubelet-serving-ca-9,sa-token-signing-certs-9 From: 18:30:58Z To: 18:30:59Z result=reject }
```

This one has me a little nervous, I can't see anything in here that would cause this, and it does happen albeit very rarely: https://prow.ci.openshift.org/view/gs/test-platform-results/logs/periodic-ci-openshift-release-master-ci-4.16-upgrade-from-stable-4.15-e2e-aws-sdn-upgrade/1770220099685322752

Main concern would be that it's something that should be allowed but no longer is due to something in this PR. We have an allowed patho for for RequiredInstallerResourcesMissing, but only for etcd operator secrets, not kube-apiserver-operator configmaps. I think the PR is probably ok but we'll re-test again.

/retest

**Comment by dgoodwin** [comment]
I don't like that this test has failed a few times on gcp-ovn:

[sig-arch] events should not repeat pathologically for ns/openshift-kube-controller-manager-operator expand_less 	0s
{  1 events happened too frequently

event happened 21 times, something is wrong: namespace/openshift-kube-controller-manager-operator deployment/kube-controller-manager-operator hmsg/7d714675bd - reason/RequiredInstallerResourcesMissing configmaps: aggregator-client-ca,client-ca, configmaps: cluster-policy-controller-config-34,config-34,controller-manager-kubeconfig-34,kube-controller-cert-syncer-kubeconfig-34,kube-controller-manager-pod-34,recycler-config-34,service-ca-34,serviceaccount-ca-34 From: 18:55:14Z To: 18:55:15Z result=reject }

however it's happening in periodics, just rarely... https://prow.ci.openshift.org/job-history/test-platform-results/logs/periodic-ci-openshift-release-master-ci-4.16-e2e-gcp-ovn-upgrade
The output is identical. We do not seem to have an exception for this. I can't see how it's this PR at present.

**Comment by dgoodwin** [comment]
/override ci/prow/e2e-metal-ipi-ovn-ipv6 

Not related to this PR.

**Pull request merged**